File: rust/src/bin/2015-04.rs
use md5;
use my_project::get_input;

fn lowest_md5(content: &str, n: usize) -> i32{
    let cmp = &"0".repeat(n);
    for i in 1.. {
        let s = format!("{}{}", content, i.to_string());
        let hash_val = format!("{:x}", md5::compute(s));
        if hash_val.starts_with(cmp){
            return i;
        }
    }
    unreachable!();
}

fn main(){
    let content = get_input(2015, 4).expect("Can't read from file");

    println!("Part A: {}", lowest_md5(&content, 5));
    println!("Part B: {}", lowest_md5(&content, 6));
}

File: python/2015-04.py
from inputGetter import get_input
import hashlib

content = get_input(2015, 4)

def lowest_md5(n: int):
    global content
    i = 1
    while True:
        s = content + str(i)
        if hashlib.md5(s.encode()).hexdigest()[:n] == '0' * n:
            break
        i += 1
    return i  

print(f'Part A: {lowest_md5(5)}')
print(f'Part B: {lowest_md5(6)}')